src/entity/Living.php

```diff
@@ -149,6 +149,25 @@ protected function initEntity(CompoundTag $nbt) : void{
 			$this->getViewers(),
 			fn(EntityEventBroadcaster $broadcaster, array $recipients) => $broadcaster->onMobArmorChange($recipients, $this)
 		)));
+		$this->armorInventory->getListeners()->add(new CallbackInventoryListener(function(Inventory $inventory, int $slot, Item $oldItem) : void{
+			$newItem = $inventory->getItem($slot);
+			if(!$newItem->isNull() && $newItem instanceof Armor && !$newItem->equalsExact($oldItem)){
+				$equipSound = $newItem->getMaterial()->getEquipSound();
+				if($equipSound !== null){
+					$this->broadcastSound($equipSound);
+				}
+			}
+		}, function(Inventory $inventory, array $oldContents) : void{
+			foreach($oldContents as $slot => $oldItem){
+				$newItem = $inventory->getItem($slot);
+				if(!$newItem->isNull() && $newItem instanceof Armor && !$newItem->equalsExact($oldItem)){
+					$equipSound = $newItem->getMaterial()->getEquipSound();
+					if($equipSound !== null){
```

Comment:
Couldn't we combine the two logic of sound broadcasting? I'm not a fan of the duplicate code that's been done here.